packages/grafana-schema/src/scuemata/dashboard/dashboard.cue

```diff
@@ -306,7 +300,29 @@ Family: scuemata.#Family & {
                             }]
                         }]
                     }
+                    // Embed the disjunction of all injected panel schema, if any were injected.
+                    if len(compose._panelSchemas) > 0 {
+                        or(compose._panelSchemas) // TODO try to stick graph in here
+                    }
+
+                    // Make the plugin-composed subtrees open if the panel is
+                    // of unknown types. This is important in every possible case:
+                    // - Base (this file only): no real dashboard json
+                    //   containing any panels would ever validate
+                    // - Dist (this file + core plugin schema): dashboard json containing
+                    //   panels with any third-party panel plugins would fail to validate,
+                    //   as well as any core plugins lacking a models.cue. The latter case
+                    //   is not normally expected, but this is not the appropriate place
+                    //   to enforce the invariant, anyway.
+                    // - Instance (this file + core + third-party plugin schema): dashboard
+                    //   json containing panels with a third-party plugin that exists but
+                    //   is not currently installed would fail to validate.
+                    if !list.Contains(compose._panelTypes, type) {
+                        options: {...}
+                        fieldConfig: defaults: custom: {...}
```

Comment:
:fire: :taco:

Reply:
Yup! Here be the magic.

It was the `list.Contains()` check - and the work necessary to make it possible - that eluded me for a while